### PR TITLE
Bug 1422968 - Combine Index Browser and Indexed Artifacts Browser

### PR DESCRIPTION
**Pull request by ydidwania**
Compared Index Browser and Indexed Artifacts Browser and added the Artifact viewing capabilities to Index Browser itself. Indexed Artifacts Browser still exists and will remove it (and make necessary changes) once these set of changes are reviewed and accepted. 
Pinging  @djmitche  for review :smiley:

### CODE REVIEW

File: src/views/IndexBrowser/EntryView.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Link } from 'react-router-dom';
 import Spinner from '../../components/Spinner';
 import DateView from '../../components/DateView';
 import JsonInspector from '../../components/JsonInspector';
+import ArtifactView from '../IndexedArtifactBrowser/ArtifactView';
```
Comment (ydidwania) [inline src/views/IndexBrowser/EntryView.js]: Will move `ArtifactView.js` to `src/IndexBrowser/` once the changes are approved.

**Comment by djmitche** [comment]
@ydidwania - we're all gathered in Austin, TX this week for lots of meetings, so we may be a little slow to review.  We will get to it!

**Comment by eliperelman** [comment]
These changes look good to me so far. Let's combine this PR with the changes to Indexed Artifacts.

**Comment by ydidwania** [comment]
@eliperelman , great, which PR is that ?
And should I delete the Indexed Artifacts Browser and push another commit?

**Comment by eliperelman** [comment]
@ydidwania yes sorry. The bug this is for specifically says that these 2 views should be combined into 1, so we need to delete the artifacts browser and only have the index browser.

**Comment by ydidwania** [comment]
I moved [ArtifactsView.js](https://github.com/taskcluster/taskcluster-tools/blob/master/src/views/IndexedArtifactBrowser/ArtifactView.js)  from `src/views/IndexedArtifactBrowser` to `src/views/IndexBrowser` . Should I also merge this ArtifactsView.js into [src/views/IndexedBrowser/EntryView.js](https://github.com/taskcluster/taskcluster-tools/blob/master/src/views/IndexBrowser/EntryView.js) ?

**Comment by eliperelman** [comment]
@ydidwania yep! any functionality that exists within the artifacts browser should make its way into the index browser.

**Comment by ydidwania** [comment]
@eliperelman , I have migrated all the functionaliy, it was just a view tbh. 
Currently the artifacts view component resides in a separate file in `src/views/IndexBrowser` . I was thinking if you wanted me to have 1 single EntryView.js instead of two separate files now.

File: src/views/IndexBrowser/EntryView.js [hunk 2]
```diff
@@ -120,6 +123,27 @@ export default class EntryView extends React.PureComponent {
...
+          <dd>
+            <ArtifactView
+              taskId={this.state.task.taskId}
+              namespace={this.state.task.namespace}
+              queue={queue}
+              userSession={userSession}
+            />
+            <br />
+            <div className="alert alert-info" role="alert">
+              <strong>Latest Artifacts</strong>&nbsp; is the artifacts from the
```
Comment (eliperelman) [inline src/views/IndexBrowser/EntryView.js]: Having a `&nbsp;` and a space right after it is putting a double space in the UI:

screen shot 2018-01-17 at 9 42 51 am

Let's remove the `&nbsp;`.

File: src/App/index.js [hunk 3]
```diff
@@ -271,10 +268,6 @@ export default class App extends React.Component {
                   path="/purge-caches"
                   component={CachePurgeInspector}
                 />
-                <PropsRoute
```
Comment (eliperelman) [inline src/App/index.js]: With this route now gone, we will need to add a redirect from this old route to the new one. You can either add a new component to handle this, or piggy-back off the `LegacyRouting` component and add a condition there to handle it:

https://github.com/taskcluster/taskcluster-tools/blob/015baa0da3bf2e7f03cb8d81eaa00575e5da2d17/src/App/LegacyRedirect.js

File: src/views/IndexBrowser/EntryView.js (hunk 2 above)
Comment (ydidwania) [reply src/views/IndexBrowser/EntryView.js]: Removing `&nbsp;` threw up a linting error 
```
 138:74  error  Insert `·last`  prettier/prettier
 139:14  error  Delete `·last`  prettier/prettier
``` 
I removed the extra space after it instead. I hope thats okay

File: src/App/index.js (hunk 3 above)
Comment (ydidwania) [reply src/App/index.js]: Added  this 
```
<PropsRoute
    path="/index/artifacts/:namespace?/:namespaceTaskId?"
    component={IndexBrowser}
/>
```

File: src/App/LegacyRedirect.js [hunk 4]
```diff
@@ -55,6 +55,10 @@ const LegacyRedirect = props => {
       );
     }
 
+    case '/index/artifacts': {
+      return <Redirect to={`/index/${props.location.pathname.slice(17)}`} />;
```
Comment (eliperelman) [inline src/App/LegacyRedirect.js]: Using a slice of this index is fragile here, we can just replace it:

```jsx
return <Redirect to={props.location.pathname.replace('/artifacts', '')} />;
```

**Comment by djmitche** [comment]
A few notes for further improvement, after this PR lands:

* On the left side, distinguish tasks from namespaces somehow.  At the moment, there's a list of tasks, a little whitespace, and then a list of namespaces.  Maybe by using a hierarchy of "twirldown" arrows like we do in the hooks manager?
* When clicking on a namespace that does not have a task, there's an alert that says "Task not found!", but this expected, so that shouldn't appear.

**Comment by eliperelman** [comment]
Thank you for your hard work on this!

**Comment by ydidwania** [comment]
Thank you too Eli for your guidance!

**Pull request merged**